### PR TITLE
[FIX] Fix performance tests

### PR DESCRIPTION
**Pull request by eddiebrissow**
Fixed

- Link generation, allowing links with strength 0
- Query using index, added indexed keyword to use index
- Commented query using 3 variables to reduce test duration

### CODE REVIEW

**Review by levisingularity — APPROVED** [review]

File: tests/performance/conftest.py [hunk 1]
```diff
@@ -19,7 +21,9 @@ def pytest_addoption(parser):
         default=0.1,
         help="Percentage of links with same letters, eg: 0.1",
     )
-    parser.addoption("--seed", default=11, help="Randon number seed")
+
+    # TODO review bounds
```
Comment (Pedrobc89) [inline tests/performance/conftest.py:25]: There is still a TODO in here, will it be addressed in another issue?

File: tests/performance/test_redis_mongo_benchmark.py [hunk 2]
```diff
@@ -129,10 +128,10 @@ def print_results(self):
         for k, v in self.test_duration.items():
             if len(v) > 1:
                 PERFORMANCE_REPORT.append(
-                    f'{k}\tAverage: {statistics.mean(v)}\tSTDEV: {statistics.stdev(v)}'
+                    f'{k}\tAverage: {statistics.mean(v) / 1000000000}\tSTDEV: {statistics.stdev(v) / 1000000000}'
```
Comment (Pedrobc89) [inline tests/performance/test_redis_mongo_benchmark.py:131]: Would rather have a constant, than a magic number, even for simple conversions like this.
Shouldn't we use the _ns() instead that already gives us time in nanoseconds?

File: tests/performance/test_redis_mongo_benchmark.py [hunk 3]
```diff
@@ -177,13 +174,16 @@ def compare_words(a: str, b: str) -> int:
...
             for j in range(i + 1, len(node_list)):
                 if random.random() > self.word_link_percentage:
                     continue
                 strength = self.compare_words(v['name'], node_list[j]['name'])
-                if strength > 0:
-                    links_dict_word[f'{i}->{j}'] = strength
+                links_dict_word[f'{i}->{j}'] = strength
+                total_links += 1
+                print(f"Word Link {total_links} created.")
+        print("Word links successfuly generated.")
```
Comment (Pedrobc89) [inline tests/performance/test_redis_mongo_benchmark.py:186]: Is this print needed, or is it a debug reminiscent?

File: tests/performance/test_redis_mongo_benchmark.py [hunk 4]
```diff
@@ -214,6 +215,10 @@ def generate_nodes(self, das: DistributedAtomSpace | None = None) -> list[dict[s
                 das.add_node(node)
             node_list.append(node)
             node_names.add(node['name'])
+            total_nodes += 1
+            print(f"Node {total_nodes} created.")
+
+        print("Nodes successfuly generated.")
```
Comment (Pedrobc89) [inline tests/performance/test_redis_mongo_benchmark.py:221]: Is this print needed, or is it a debug reminiscent?

File: tests/performance/test_redis_mongo_benchmark.py [hunk 5]
```diff
@@ -253,14 +258,17 @@ def generate_links_letter(self, node_list: list[dict[str, Any]]) -> dict[str, An
...
                 if random.random() > self.letter_link_percentage:
                     continue
                 key = f'{min(i, j)}->{max(i, j)}'
                 strength = TestPerformance.compare_str(node_list[i]['name'], node_list[j]['name'])
-                if strength > 0:
-                    links_letter[key] = strength
+                links_letter[key] = strength
+                total_links += 1
+                print(f"Letter Link {total_links} created.")
+        print("Letter links successfuly generated.")
```
Comment (Pedrobc89) [inline tests/performance/test_redis_mongo_benchmark.py:271]: Is this print needed, or is it a debug reminiscent?

**Review by Pedrobc89 — COMMENTED** [review]
There is a lot of new prints, are they all needed?

File: tests/performance/conftest.py (hunk 1 above)
Comment (andre-senna) [reply tests/performance/conftest.py:25]: It's OK if not, but please create a card for any TODO left in the code. Actually this is a general guideline: every TODO in code is supposed to be added as a card as well.

**Review by andre-senna — APPROVED** [review]

File: tests/performance/conftest.py (hunk 1 above)
Comment (eddiebrissow) [reply tests/performance/conftest.py:25]: Issue created https://github.com/singnet/das-query-engine/issues/353

**Pull request merged**